pkg/command/service/generate.go

```diff
@@ -0,0 +1,186 @@
+package service
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+
+	knativeapis "knative.dev/pkg/apis"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	servingv1client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+
+	"github.com/spf13/cobra"
+
+	"github.com/zhanggbj/kperf/pkg"
+	"github.com/zhanggbj/kperf/pkg/generator"
+)
+
+var (
+	count, interval, batch, concurrency, minScale, maxScale int
+	nsPrefix, nsRange, ns                                   string
+	svcNamePrefixDefault                                    string = "testksvc"
+	checkReady                                              bool
+	timeout                                                 time.Duration
+	ksvcClient                                              *servingv1client.ServingV1Client
+	err                                                     error
+)
+
+func NewServiceGenerateCommand(p *pkg.PerfParams) *cobra.Command {
+	ksvcGenCommand := &cobra.Command{
+		Use:   "generate",
+		Short: "generate ksvc",
+		Long: `generate ksvc workload
+
+For example:
+# To generate ksvc workload
+kperf service generate —count 500 —interval 20 —batch 20 --min-scale 0 --max-scale 5 (--nsprefix testns/ --ns nsname)
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var nsRangeMap map[string]bool = map[string]bool{}
+			if nsPrefix != "" {
+				r := strings.Split(nsRange, ",")
+				if len(r) != 2 {
+					fmt.Printf("Expected Range like 1,500, given %s\n", nsRange)
+					os.Exit(1)
+				}
+				start, _ := strconv.Atoi(r[0])
+				end, _ := strconv.Atoi(r[1])
+				if start > 0 && end > 0 && start <= end {
+					for i := start; i <= end; i++ {
+						nsRangeMap[fmt.Sprintf("%s-%d", nsPrefix, i)] = true
```

Comment:
As mentioned above, it should match the `nsRangeMap` in clean.go.

Reply:
@gyohuangxin nice catch, thanks! This will be refined in https://github.com/knative-sandbox/kperf/issues/4 after the initial PR merged.

Reply:
Get it, thanks!